### PR TITLE
Object fit video polyfill for IE and Edge 

### PR DESCRIPTION
**Pull request by rich-nguyen**
## What does this change?
Adds a polyfill for `object-fit` currently used in fabric videos.

### CODE REVIEW

**Review by JonNorman — APPROVED** [review]

**Comment by sndrs** [comment]
fyi @SiAdcock @GHaberis

File: static/src/stylesheets/module/commercial/creatives/_fabric-video.scss [hunk 1]
```diff
@@ -19,6 +19,9 @@
 
     .creative__video {
         object-fit: cover;
+
+        // this font-family style is for use with the 'object-fit-videos' polyfill.
+        font-family: 'object-fit: cover;';
```
Comment (sndrs) [inline static/src/stylesheets/module/commercial/creatives/_fabric-video.scss]: eurgh! does width/height = 100% not work?

**Comment by sndrs** [comment]
how big is the polyfill? how does it detect the presence of the css property? does it need to trigger layout to detect elements have it applied?

**Comment by SiAdcock** [comment]
Is it not acceptable to treat `object-fit: cover` as a progressive enhancement, rather than a must-have for all browsers?

File: static/src/stylesheets/module/commercial/creatives/_fabric-video.scss (hunk 1 above)
Comment (rich-nguyen) [reply static/src/stylesheets/module/commercial/creatives/_fabric-video.scss]: This hack is used by the polyfill module. The intersection of 'polyfill that works' and 'polyfill with reasonable modular structure' is very small.

**Comment by rich-nguyen** [comment]
> how big is the polyfill?
3KB minified.
> Is it not acceptable to treat object-fit: cover as a progressive enhancement?
No. The campaign is already sold, so the client is expecting this ad format to work on their browser. The campaign is going live imminently, hence the quick fix. Don't worry though, @sndrs and @SiAdcock! we have a DFP native style version of this format ready to go. When we migrate to the native version, all of this code that you are looking at disappears from Frontend, and the ad content will be served exclusively when the creative is selected by the ad server.

**Comment by sndrs** [comment]
if we've already sold it and now its about to go live, i don't really see that we can raise an objection. 

but we should be really careful that we're selling things we feel ok about building and running on our site. adding polyfills, _esp_ for CSS properties, gives me major shivers (in particular when we're  firefighting degraded performance already)

do you know how long till the native version kicks in?

**Comment by sndrs** [comment]
i think it will trigger layout https://github.com/TricomB2B/object-fit-videos/blob/master/object-fit-videos.js#L56

can you fastdom it? then if we can set a date to remove it, 👍

**Comment by rich-nguyen** [comment]
I don't think `getComputedStyle` will trigger a layout for a `font-family` property, just a style calc. According to Paul: https://gist.github.com/paulirish/5d52fb081b3570c81e3a#getcomputedstyle

I'll certainly add a date, thanks for the feedback @sndrs

**Comment by sndrs** [comment]
it will trigger layout here i think? https://github.com/TricomB2B/object-fit-videos/blob/master/object-fit-videos.js#L108-L126

**Comment by rich-nguyen** [comment]
That function is wrapped in a fastdom write.

**Comment by sndrs** [comment]
shouldn't the `getComputedStyle` there be in a `read`, ideally?

**Comment by rich-nguyen** [comment]
Only if the library exposed the function, to allow us to separate the interleaved read and writes. Given that I can't, leaving a write in fastdom's read queue seems quite problematic.

**Comment by SiAdcock** [comment]
Thanks for the clarifications, @rich-nguyen. Serving 3KB of JS to everyone in order to support a particular ad in Edge was initially concerning, but it's reassuring to know this is only a temporary solution 😌

**Pull request merged**